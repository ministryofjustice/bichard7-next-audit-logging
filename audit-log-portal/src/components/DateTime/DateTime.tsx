import { format } from "date-fns"

interface Props {
  date?: Date | string
  prefix?: string
}

const DateTime = ({ date, prefix }: Props) => {
  if (!date) {
    return <></>
  }

  const dateObject = new Date(date)

  return (
    <>
      {prefix}
      <time aria-label="time">{format(dateObject, "dd/MM/yyyy HH:mm:ss")}</time>
    </>
  )
}

export default DateTime
