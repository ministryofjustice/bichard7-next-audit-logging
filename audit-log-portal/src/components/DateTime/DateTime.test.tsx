import { render, screen } from "@testing-library/react"
import DateTime from "./DateTime"

test("matches snapshot", () => {
  const date = new Date("2021-05-21T08:27")

  const { container } = render(<DateTime date={date} />)

  expect(container).toMatchSnapshot()
})

test("has correct date time format when date is type of Date", () => {
  const date = new Date("2021-04-01T13:20:30")
  const expectedDateString = "01/04/2021 13:20:30"

  render(<DateTime date={date} />)

  const time = screen.getByLabelText("time")
  expect(time.innerHTML).toBe(expectedDateString)
})

test("has correct date time format when date is type of String", () => {
  const dateString = "2021-04-01T13:20:30"
  const expectedDateString = "01/04/2021 13:20:30"

  render(<DateTime date={dateString} />)

  const time = screen.getByLabelText("time")
  expect(time.innerHTML).toBe(expectedDateString)
})

test("not rendered when date has no value", () => {
  render(<DateTime date={null} />)

  const time = screen.queryByLabelText("time")
  expect(time).toBeNull()
})
