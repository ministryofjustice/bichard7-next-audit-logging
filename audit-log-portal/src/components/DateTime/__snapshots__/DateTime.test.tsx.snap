// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <time
    aria-label="time"
  >
    21/05/2021 08:27:00
  </time>
</div>
`;
